src/main/java/com/cloudbees/jenkins/plugins/advisor/casc/AdvisorRootConfigurator.java

```diff
@@ -0,0 +1,151 @@
+
+    private static final Logger LOG = Logger.getLogger(AdvisorRootConfigurator.class.getName());
+
+    // Ignoring lastBundleResult since it is not configured in the plugin. It only informs about the last bundle generation
+    // Ignoring isValid since it is something auto-calculated during the configuration
+    private final Collection<String> excludedAttributesInConf = Arrays.asList("lastBundleResult", "valid");
+
+    @NonNull
+    @Override
+    public String getName() {
+        return "advisor";
+    }
+
+    @Override
+    public String getDisplayName() {
+        return getConfiguration().getDisplayName();
+    }
+
+    @Override
+    protected AdvisorGlobalConfiguration instance(Mapping mapping, ConfigurationContext configurationContext) throws ConfiguratorException {
+        // Scalar values
+        final String email = (mapping.get("email") != null ? mapping.getScalarValue("email") : StringUtils.EMPTY);
+        final String cc = (mapping.get("cc") != null ? mapping.getScalarValue("cc") : StringUtils.EMPTY);
+        final boolean nagDisabled = (mapping.get("nagDisabled") != null ? BooleanUtils.toBoolean(mapping.getScalarValue("nagDisabled")) : false);
+        final boolean acceptToS = (mapping.get("acceptToS") != null ? BooleanUtils.toBoolean(mapping.getScalarValue("acceptToS")) : false);
+
+        // List values
+        final Set<String> excludedComponents = new HashSet<>();
+        CNode excludedCN = mapping.get("excludedComponents");
+        if (excludedCN != null) {
+            if (excludedCN instanceof Sequence) {
+                Sequence s = (Sequence) excludedCN;
+                for (CNode cNode : s) {
+                    excludedComponents.add(cNode.asScalar().getValue());
+                }
+
+                if (excludedComponents.isEmpty()) {
+                    excludedComponents.add(AdvisorGlobalConfiguration.SEND_ALL_COMPONENTS);
+                }
+            } else {
+                throw new ConfiguratorException(this, "Excluded components are expected to be a list.");
+            }
+
+        } else {
+            excludedComponents.add(AdvisorGlobalConfiguration.SEND_ALL_COMPONENTS);
+        }
+
+        // Check and apply configuration
+        if (!acceptToS) {
+            // In UI, if you don't accept the ToS, the configuration is not applied, so here we throw an exception
+            throw new ConfiguratorException(this, "Terms of Service for CloudBees Jenkins Advisor have to be accepted. Please, review the acceptToS field in the yaml file.");
+        }
+
+        AdvisorGlobalConfiguration insights = getTargetComponent(configurationContext);
+        AdvisorGlobalConfiguration.DescriptorImpl descriptor = (AdvisorGlobalConfiguration.DescriptorImpl) insights.getDescriptor();
+        boolean valid = true;
+        if (descriptor.doCheckEmail(email).kind.equals(FormValidation.Kind.ERROR) || descriptor.doCheckCc(cc).kind.equals(FormValidation.Kind.ERROR)) {
+            valid = false;
```

Comment:
You could throw here the exception and update the configuration in an else and you save the variable.